sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeUtils.scala

```diff
@@ -55,10 +56,19 @@ object DateTimeUtils {
   // this is year -17999, calculation: 50 * daysIn400Year
   final val YearZero = -17999
   final val toYearZero = to2001 + 7304850
-  final val TimeZoneGMT = TimeZone.getTimeZone("GMT")
 
   @transient lazy val defaultTimeZone = TimeZone.getDefault
 
+  // Reuse the TimeZone object as it is expensive to create in each method call.
+  final val timeZones = new ConcurrentHashMap[String, TimeZone]
```

Comment:
This map could be quite big, because the string varies. Actually `ZoneInfoFile` does provide a cache for different `ID`s. Let's find out whether the boost you mentioned comes from reusing `TimeZone` or `Calendar` instances.